system-int-test/test/cmr/system_int_test/search/data_json_search_test.clj

```diff
@@ -0,0 +1,92 @@
+  "This namespace contains tests for the data.json endpoint."
+  (:require
+   [clojure.test :refer :all]
+   [cmr.common-app.test.side-api :as side-api]
+   [cmr.mock-echo.client.echo-util :as echo-util]
+   [cmr.search.api.concepts-search :as concepts-search]
+   [cmr.system-int-test.data2.core :as core]
+   [cmr.system-int-test.data2.umm-spec-collection :as umm-spec-collection]
+   [cmr.system-int-test.system :as system]
+   [cmr.system-int-test.utils.index-util :as index-util]
+   [cmr.system-int-test.utils.ingest-util :as ingest-util]
+   [cmr.system-int-test.utils.search-util :as search-util]
+   [cmr.system-int-test.utils.tag-util :as tag-util]
+   [cmr.transmit.config :as config]))
+
+(def ^:private coll-idx (atom 0))
+
+(defn- data-json-fixture
+  "Enable/Disable data.json endpoint, reset coll-idx for collection creation."
+  [f]
+  (let [saved-data-json-flag (concepts-search/allow-data-json-flag)]
+    (side-api/eval-form `(concepts-search/set-allow-data-json-flag! true))
+    (reset! coll-idx 0)
+    (f)
+    (side-api/eval-form `(concepts-search/set-allow-data-json-flag! ~saved-data-json-flag))))
+
+(use-fixtures :each (join-fixtures
+                      [(ingest-util/reset-fixture {"provguid1" "PROV1"}
+                                                  {:grant-all-search? false})
+                       tag-util/grant-all-tag-fixture
+                       data-json-fixture]))
+
+(defn- create-collections
+  "Create num-collections collections. Default provider is PROV1."
+  ([num-collections]
+   (create-collections num-collections "PROV1"))
+  ([num-collections provider]
+   (let [start-idx @coll-idx
+         end-idx (+ start-idx num-collections)]
+     (reset! coll-idx end-idx)
+     (doall
+      (for [idx (range start-idx end-idx)]
+        (core/ingest-umm-spec-collection
+         provider
+         (umm-spec-collection/collection idx {})))))))
+
+(defn- make-collections-public
+  "Take collection of umm-spec collections and make public. Default provider is PROV1."
+  ([collections]
+   (make-collections-public collections "PROV1"))
+  ([collections provider]
+   (->> (mapv :EntryTitle collections)
+        echo-util/coll-id
+        (echo-util/coll-catalog-item-id provider)
+        (echo-util/grant-guest (system/context)))))
+
+(defn- associate-collections
+  "Associate collections with tag."
+  [tag collections]
+  (tag-util/associate-by-concept-ids
+   (config/echo-system-token)
+   (:tag-key tag)
+   (map #(hash-map :concept-id (:concept-id %)) collections)))
+
+(deftest data-json-response
+  (let [tag (tag-util/make-tag {:tag-key "gov.nasa.eosdis"})
+        public-collections-with-tag (create-collections 7)
+        public-collections-without-tag (create-collections 6)
+        private-collections-with-tag (create-collections 5)
+        private-collections-without-tag (create-collections 4)]
+    (tag-util/create-tag (config/echo-system-token) tag)
+    (index-util/wait-until-indexed)
```

Comment:
This line can be removed as the wait done on line 79 is enough to make sure tags are available before association.